doc/policy_based_metering/PBM-HLD.md

```diff
@@ -0,0 +1,352 @@
+
+---
+### SAI API
+| SAI Attribute                           | Description                      |
+| --------------------------------------- | -------------------------------- |
+| set_acl_entry_attribute()               | Attach policers to ACL rules     |
+| sai_query_attribute_capability()        | Verify ASIC support for policers |
+| SAI_ACL_ACTION_TYPE_SET_POLICER         |                                  |
+| SAI_ACL_ACTION_TYPE_PACKET_ACTION       |                                  |
+| SAI_ACL_ENTRY_ATTR_ACTION_PACKET_ACTION |                                  |
+| SAI_ACL_ENTRY_ATTR_ACTION_SET_POLICER   |                                  |
+
+---
+### Configuration and Management
+
+#### Config DB Enhancements
+
+##### Updated ACL Tables Table schema:
+- The current SAI implementation requires an action list when a new ACL is created.
+  Until this feature was introduced, only default actions were supported, so SONiC addressed this by sending a default action list (**defaultAclActionList**) per table type.
+- Since a new action is added, ACL Tables Table schema will be updated with a new optional action-list field - **ext_actions_list** to accept list of actions, these actions in addition to defaultAclActionList will be used to creating ACL table.
+- The list of actions will be augmented to the existing defaultAclActionList.
+  For example, if user configures **ext_actions_list**=**POLICER**, **ACTION1** - the full action-list of that table will be \[\<**defaultAclActionList(table_type)**\>, **POLICER**, **ACTION1**\]
+```
+  key                  = ACL_TABLE:name                            ; acl_table_name must be unique
+  ;field               = value                         
+  policy_desc          = 1*255VCHAR                                ; name of the ACL policy table description
+  type                 = 1*255VCHAR                                ; type of acl table, every table type
+                                                                   ; defines a specific set of actions.
+                                                                   ; posible types:
+                                                                   ; "L3"/"L3V6"/"L3V4V6"/"MIRROR"/"MIRROR_DSCP"/
+                                                                   ; "MIRRORV6"/"PFCWD"/"MCLAG"/"MUX"/"DROP"
+  ports                = [0-max_ports]*port_name                   ; the ports to which this ACL
+                                                                   ; table is applied, can be emtry
+                                                                   ; value annotations
+  port_name            = 1*64VCHAR                                 ; name of the port, must be unique
+  max_ports            = 1*5DIGIT                                  ; number of ports supported on the chip
+* ext_actions_list     = ["POLICER", "ACTION1", "ACTION2", ...]    ; additional action list - can be extended in the future with new actions
```

Comment:
Currently only POLICER, right? I'd remove the not implemented ones (ACTION*)
Where is the mapping of the name POLICER to the policer table
empty is default? it always happens? for backward compatibility

Reply:
Empty is the default - this list defines the actions that the ACL rules can be used